batch/batch.go

```diff
@@ -0,0 +1,134 @@
+package batch
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/pokt-foundation/transaction-db/types"
+	"github.com/sirupsen/logrus"
+	"golang.org/x/sync/errgroup"
+)
+
+type RelayWriter interface {
+	WriteRelays(ctx context.Context, relays []types.Relay) error
+}
+
+type PublicWriter interface {
+	WriteRelays(ctx context.Context, relays []types.Relay) error
+}
+
+type Batch struct {
+	relays      []types.Relay
+	rwMutex     sync.RWMutex
+	batchChan   chan types.Relay
+	maxSize     int
+	maxDuration time.Duration
+	timeoutDB   time.Duration
+	writer      RelayWriter
+	log         *logrus.Logger
+}
+
+func (b *Batch) logError(err error) {
+	fields := logrus.Fields{
+		"err": err.Error(),
+	}
+
+	b.log.WithFields(fields).Error(err)
+}
+
+func NewBatch(maxSize int, maxDuration, timeoutDB time.Duration, writer RelayWriter, logger *logrus.Logger) *Batch {
```

Comment:
function could simply be called `New` that way when is called from any other package is read like `batch.New` not `batch.NewBatch`